src/types/libp2p.ts

```diff
@@ -0,0 +1,60 @@
+import { bytes, bytes32 } from "./basic";
+import { NoiseSession } from "../xx";
+
+export interface KeyPair {
+  publicKey: bytes32,
+  privateKey: bytes32,
+}
+
+export type PeerId = {
+  id: string,
+  privKey: string,
+  pubKey: string,
+};
+
+type PeerInfo = {
+  noiseKey: bytes32,
+  libp2pKey: bytes,
+};
+
+type ConnectionStats = {
+  direction: "inbound" | "outbound",
+  encryption: string,
+}
+
+
+// Also seen as Pair
+export type Stream = {
+  sink(source: Iterable<any>),
+  source: Object,
+}
+
+export type Duplex = [Stream, Stream];
+
+export interface InsecureConnection {
+  localPeer: PeerId,
+  remotePeer: PeerId,
+  local: PeerInfo,
+  remote: PeerInfo,
+  stats: ConnectionStats,
+
+  streams(): Duplex,
+  addStream(muxedStream: any) : Stream,
+}
+
+export interface NoiseConnection {
+  remoteEarlyData?(): bytes,
+  secureOutbound(insecure: InsecureConnection, remotePeer: PeerId): Promise<SecureConnection>,
+  secureInbound(insecure: InsecureConnection): Promise<SecureConnection>,
```

Comment:
I don't think that's correct types.
Secio returns just {conn, remotePeer}

Reply:
Mhm, will update according to https://github.com/libp2p/js-interfaces/tree/master/src/crypto#ap